tests/Integration/inc/classes/subscriber/Media/WebpSubscriber/convertToWebp.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace WP_Rocket\Tests\Integration\inc\classes\subscriber\Media\WebpSubscriber;
+
+use Brain\Monkey\Functions;
+use WPMedia\PHPUnit\Integration\TestCase;
+
+/**
+ * @covers \WP_Rocket\Subscriber\Media\Webp_Subscriber::convert_to_webp
+ * @group Subscriber
+ * @group WebP
+ */
+class Test_ConvertToWebp extends TestCase {
+	/**
+	 * @dataProvider noMatchProvider
+	 */
+	public function testShouldReturnHtmlWithCommentWhenNoMatches( $original, $expected ) {
+		add_filter( 'pre_get_rocket_option_cache_webp', [ $this, 'return_1' ] );
+
+		Functions\when( 'apache_request_headers' )
+			->alias( function() {
+				return [
+					'Accept' => 'webp',
+				];
+			}
+		);
+
+		$this->assertSame( $expected, apply_filters( 'rocket_buffer', $original ) );
+
+		remove_filter( 'pre_get_rocket_option_cache_webp', [ $this, 'return_1' ] );
+	}
+
+	public function noMatchProvider() {
+		return $this->getTestData( __DIR__, 'convert-to-webp-nomatch' );
```

Comment:
@Tabrisrp How about an integration test for when it should convert into webp? Can you combine both test data sets into one to load all of it into the data provider? In that way, we are testing the behavior with different states to check what happens.

Reply:
The integration test for successful conversion to WebP will require to use the virtual filesystem.

I kept it simple by checking the issue at hand specifically here, I can see if it's doable to check for this in addition to it, but it's possible I will get into blockers

Reply:
You did it right: write the test for this specific issue. If the other state is doable, cool let's do it. Else, we can do it when we touch that code.